tests/ttnn/unit_tests/benchmarks/test_benchmark.py

```diff
@@ -66,3 +71,217 @@ def test_benchmark_ttnn_matmul(device, use_program_cache, m_size, k_size, n_size
+                layout=ttnn.TILE_LAYOUT,
+                device=device,
+                memory_config=ttnn.DRAM_MEMORY_CONFIG,
+            )
+
+            program_config = ttnn.MatmulMultiCoreReuseMultiCastProgramConfig(
+                compute_with_storage_grid_size=grid_size,
+                in0_block_w=in0_block_w,
+                out_subblock_h=out_subblock_h,
+                out_subblock_w=out_subblock_w,
+                per_core_M=out_block_h,
+                per_core_N=out_block_w,
+                transpose_mcast=False,
+                fused_activation=None,
+            )
+
+            if is_grayskull():
+                compute_kernel_config = ttnn.GrayskullComputeKernelConfig(
+                    math_fidelity=math_fidelity,
+                    math_approx_mode=True,
+                )
+            else:
+                compute_kernel_config = ttnn.WormholeComputeKernelConfig(
+                    math_fidelity=math_fidelity,
+                    math_approx_mode=True,
+                    fp32_dest_acc_en=False,
+                    packer_l1_acc=True,
+                )
+            if out_sharded:
+                out_mem_config = ttnn.MemoryConfig(
+                    memory_layout=ttnn.TensorMemoryLayout.BLOCK_SHARDED,
+                    buffer_type=ttnn.BufferType.L1,
+                )
+            else:
+                out_mem_config = ttnn.DRAM_MEMORY_CONFIG
+            if out_sharded:
+                output_tile = ttnn.Tile([tile_h, 32]) if tile_h <= 16 else ttnn.Tile([tile_h, tile_w])
+            else:
+                output_tile = ttnn.Tile([tile_h, tile_w])
+
+            output_t = ttnn.matmul(
+                in0_t,
+                in1_t,
+                program_config=program_config,
+                memory_config=out_mem_config,
+                dtype=dtype,
+                compute_kernel_config=compute_kernel_config,
+                output_tile=output_tile,
+            )
+
+            tid = ttnn.begin_trace_capture(device, cq_id=0)
+            output_t = ttnn.matmul(
+                in0_t,
+                in1_t,
+                program_config=program_config,
+                memory_config=out_mem_config,
+                dtype=dtype,
+                compute_kernel_config=compute_kernel_config,
+                output_tile=output_tile,
+            )
+            ttnn.end_trace_capture(device, tid, cq_id=0)
+
+            for iter in range(0, num_warmup_iterations):
+                ttnn.execute_trace(device, tid, cq_id=0, blocking=False)
+
+            profiler.start(f"run")
+            for iter in range(0, num_measurement_iterations):
+                ttnn.execute_trace(device, tid, cq_id=0, blocking=False)
+            ttnn.synchronize_device(device)
+            profiler.end(f"run")
+            ttnn.DumpDeviceProfiler(device)
+            inference_time_avg = profiler.get("run") / num_measurement_iterations
+            tflops = 2 * m * k * n / 1e12 / inference_time_avg
```

Comment:
This is assuming WH clock speed? I would put these into sensible constants

Reply:
no clk speed assumed, 1e12 is to get tflops from flops

Reply:
Correct, and the factor of 2 is because we get 2 operations from an FMA